DiagnosticsExtension/Models/ConnectionStringValidator/Exceptions/ConnectionStringResponseUtility.cs

```diff
@@ -9,58 +9,57 @@
+        public const string System = "System";
         public const string ClientId = "__clientId";
         public const string Credential = "__credential";
-        public const string ServiceUriMissed = "ServiceUriMissed";
+        public const string ServiceUri = "__serviceUri";
+        public const string BlobServiceUri = "__blobServiceUri";
+        public const string ServiceUriMissing = "ServiceUriMissing";
+        public const string QueueServiceUri = "__queueServiceUri";
         public const string ValidCredentialValue = "managedidentity";
         public const string FullyQualifiedNamespace = "__fullyQualifiedNamespace";
         public const string ManagedIdentityCredentialMissing = "ManagedIdentityCredentialMissing";
         #endregion
 
-        public static ConnectionStringValidationResult EvaluateResponseStatus(Exception e, ConnectionStringType type, ConnectionStringValidationResult.ManagedIdentityType identityType)
+        public static void EvaluateResponseStatus(Exception e, ConnectionStringType type, ref ConnectionStringValidationResult response)
         {
-            var response = new ConnectionStringValidationResult(type);
             if (e is MalformedConnectionStringException)
             {
                 response.Status = ConnectionStringValidationResult.ResultStatus.MalformedConnectionString;
             }
-            else if (e.Message.Contains("managedIdentityCredentialMissing"))
+            else if (e is EmptyConnectionStringException)
             {
-                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredentialMissing;
+                response.Status = ConnectionStringValidationResult.ResultStatus.EmptyConnectionString;
             }
-            else if (e.Message.Contains("Unauthorized") || e.Message.Contains("AuthorizationPermissionMismatch"))
+            else if (e.Message.Contains(ManagedIdentityCredentialMissing))
             {
-                if (identityType == ConnectionStringValidationResult.ManagedIdentityType.User)
-                {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.UserAssignedManagedIdentity;
-                }
-                else
-                {
-                    response.Status = ConnectionStringValidationResult.ResultStatus.SystemAssignedManagedIdentity;
-                }
+                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredentialMissing;
             }
-            else if (e.Message.Contains("ManagedIdentityCredential"))
+            else if (e.Message.Contains("Unauthorized") || e.Message.Contains("unauthorized") || e.Message.Contains("request is not authorized"))
             {
-                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredential;
+                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityAuthFailure;
```

Comment:
This looks like the wrong result status code

Reply:
fixed